Cabal/Distribution/Simple/Setup.hs

```diff
@@ -703,6 +707,13 @@ configureOptions showOrParseArgs =
          configUseResponseFiles
          (\v flags -> flags { configUseResponseFiles = v })
          (boolOpt' ([], ["disable-response-files"]) ([], []))
+
+      ,option "" ["allow-depending-on-private-libs"]
```

Comment:
This should be added to `filterConfigureFlags` as well, though since we don't set it anywhere, it's not a blocker for merging.